### DISCUSSION TITLE
No such file or directory: '/srv/nominatim/website' after reboot

### DISCUSSION THREAD

**Discussion by echo292** [osm-search/Nominatim#3231] [Q&A]
Followed the install instructions exactly to the documentation. 
After install and running "nominatim sever" I was able to make request through the php api.
After rebooting I ran the apache2 server with the following command and I get the error listed below. 

```
sudo a2enconf nominatim
sudo systemctl restart apache2
```

```

nominatim@Ubuntu22v2:~$ nominatim serve
 Using project directory: /srv/nominatim
Traceback (most recent call last):
  File "/usr/local/bin/nominatim", line 12, in <module>
    exit(cli.nominatim(module_dir='/usr/local/lib/nominatim/module',
  File "/usr/local/lib/nominatim/lib-python/nominatim/cli.py", line 225, in nominatim
    return get_set_parser().run(**kwargs)
  File "/usr/local/lib/nominatim/lib-python/nominatim/cli.py", line 121, in run
    return args.command.run(args)
  File "/usr/local/lib/nominatim/lib-python/nominatim/cli.py", line 169, in run
    run_php_server(args.server, args.project_dir / 'website')
  File "/usr/local/lib/nominatim/lib-python/nominatim/tools/exec_utils.py", line 25, in run_php_server
    subprocess.run(['/usr/bin/env', 'php', '-S', server_address],
  File "/usr/lib/python3.10/subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.10/subprocess.py", line 1863, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/srv/nominatim/website'
```

 I also confirmed the /etc/apache2/conf-available/nominatim.conf looks correct. 

```
<Directory "/srv/nominatim/nominatim-project/website">
  Options FollowSymLinks MultiViews
  AddType text/html   .php
  DirectoryIndex search.php
  Require all granted
</Directory>

Alias /nominatim /srv/nominatim/nominatim-project/website
```

**Comment by mtmail** [comment]
> nominatim@Ubuntu22v2:~$ nominatim serve
> Using project directory: /srv/nominatim

Your project directory is /srv/nominatim/nominatim-project

Either first change into the directory (cd) or add a command-line parameter (nominatim serve -h will show the syntax).

It helps if you list which version of Nominatim you installed and which tutorial or documentation page you followed, there’s multiple.

**Reply by echo292** [reply]
Hello, apologies for the limited info. Thank you for your assistance. This is my first round exploring Nominatim. 
Nominatim version 4.3.1-0

Documentation:
- https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-22/
- https://nominatim.org/release-docs/latest/admin/Import/

As per your feedback, I realized I was running the serve command in the home folder.  

I'm now able to execute API calls on the command. 

I have the php server running however I'm getting a GET /status.php  -  No such file or directory

Thank you for your help.

-E

**Comment by echo292** [comment]
Is it necessary to run the apache2 server then the nominatim server separate?
Or will the "nominatim server --engine php" be enough?

I've tried both ways however I still get the following message. 
GET /status.php - No such file or directory

If they both need to run, which one runs on the loopback and which will run on a separate ip?

When I run the "netstat -tnlp" I see no ip assigned to port 80. 

Thank you,

-E

**Comment by echo292** [comment]
Disregard my questions above. I figured it out. 
Closing comment.

**Comment by echo292** [comment]
closing
